### DISCUSSION TITLE
upstream TS data by the Datapoint Name not by the ID

### DISCUSSION THREAD

**Discussion by IlyaMatsuk** [mindsphere/node-red-contrib-mindconnect#115] [General]
Hello. Recently I have started to use Node-RED agent. Do we have opportunity to upstream TS data by the Datapoint Name not by the ID?

**Comment by sn0wcat** [comment]
Hi Ilya,

Unfortunately the MindSphere requires data point ids and not names. However you can use IDs like "DP-var1", "DP-var2" etc. if you use the configuration dialog in node to configure the dialog, the node will use [this algorithm](https://github.com/mindsphere/mindconnect-nodejs/blob/2afbbdf4bc04a245e44c11aaad19e3e38cfadd5b/src/api/sdk/mcapi/mcapi.ts#L368) to generate the ids. 

There is also an [article](https://sn0wcat-92713.medium.com/how-to-create-mindsphere-artefacts-from-json-schema-definitions-1b2f3e446f6a) and [example](https://playground.mindconnect.rocks/#flow/996ec513.e1f4d8) how to generate the ids from existing data.

(the "tricks" are there because you can't always name the ids DP-variable name because of mindsphere restrictions (the ids must be unique per asset, restricted to 36 characters in length etc.)

**Reply by IlyaMatsuk** [reply]
Thanks for the answer and article. There are very useful!

**Reply by IlyaMatsuk** [reply]
In the article in "Step 0: Setup the MindSphere CLI", you configure the MindSphere CLI using Developer Cockpit but in my case, I believe it works for Dev tenant but I have to work with assets and datapoints already at Value Plan Tenant. Please advise how can I do this?

**Reply by sn0wcat** [reply]
Either you can use the token from your op tenant or you can borrow the cookies from e.g. asset manager or fleet manager of the IOT tenant.  

see also https://opensource.mindsphere.io/docs/mindsphere-auth-helper/index.html#tab1anchor2

**Reply by IlyaMatsuk** [reply]
Thanks. I have done step0 and 1 from the [article](https://sn0wcat-92713.medium.com/how-to-create-mindsphere-artefacts-from-json-schema-definitions-1b2f3e446f6a). But when I try to execute the command "mc aspects --mode create --file RobotSensorData.aspect.mdsp.json" in the "Step 2 Create Aspects in MindSphere", I receive the next error: 
"Network error occured Bad Request {"errors":[{"code":"assetmngmnt.aspecttype.notowned.change","logref":"6022a3d4f19f63c8e5c1c72e43072c34","message":"Aspect type with id %s does not belong to user's tenant, therefore cannot be changed.","errorcode":"assetmngmnt.aspecttype.notowned.change"}]}"
Could you help me to understand the cause of the error?
![dsBuffer bmp](https://user-images.githubusercontent.com/46844189/107383628-38ae8480-6b02-11eb-8670-e856f4097270.png)

**Reply by sn0wcat** [reply]
@IlyaMatsuk you could try to edit the file and write the full aspect name e.g. youriottenant.RobotSensorData instead of RobotSensorData in the name property.

**Reply by IlyaMatsuk** [reply]
I used the full aspect name and now I have the next error:
"Network error occured Bad Request {"errors":[{"code":"assetmngmnt.aspecttype.name.pattern","logref":"6022ab51fd286be562a0291b7b28ca0b","message":"Given name is not a valid aspect type name","errorcode":"assetmngmnt.aspecttype.name.pattern"}]}"

**Reply by IlyaMatsuk** [reply]
> I used the full aspect name and now I have the next error:
> "Network error occured Bad Request {"errors":[{"code":"assetmngmnt.aspecttype.name.pattern","logref":"6022ab51fd286be562a0291b7b28ca0b","message":"Given name is not a valid aspect type name","errorcode":"assetmngmnt.aspecttype.name.pattern"}]}"

Hello, @sn0wcat. Do you have an idea about how I can fix this? Thanks

**Reply by sn0wcat** [reply]
I would have to test it myself with an IoT value plan to see what exactly is going wrong... it will probably take a few days, this is unfortunately not my day job ;)

**Reply by IlyaMatsuk** [reply]
> I would have to test it myself with an IoT value plan to see what exactly is going wrong... it will probably take a few days, this is unfortunately not my day job ;)

Hello, @sn0wcat. Thank you for the answer and help. I'll wait for the result of your test.

**Reply by IlyaMatsuk** [reply]
Hello, @sn0wcat. Can I help you with this problem? Maybe we can check this and fix it together?!
